### ISSUE TITLE
IllegalArgumentException with Java 9

### ISSUE THREAD

**Issue by HoldYourWaffle** [MutabilityDetector/MutabilityDetector#110]
So I tried my very first mutability test using JUnit 4.12, but it failed horribly.
[Pastebin of test report](https://pastebin.com/JbX6GHcQ)

Test and the analyzed class:
```java
public class ImmutabilityTest {
	@Test
	public void verify() {
		MutabilityAssert.assertImmutable(MethodInvoker.class);
	}
}
```

```java
public class MethodInvoker {
	public final Method method;
	public final Object target;
	
	public MethodInvoker(Method method, Object target) {
		this.method = method;
		this.target = target;
	}
}
```

This almost hello world-like test should've failed (`MethodInvoker` is not final), but instead of failing the assertion it 'crashed' with an `IllegalArgumentException` (wrapped in a `MutabilityAnalysisException`) and a clear message saying I should report this.
##### _hereby the report_

Other (maybe) useful information:
```
------------------------------------------------------------
Gradle 4.4.1
------------------------------------------------------------

Build time:   2017-12-20 15:45:23 UTC
Revision:     10ed9dc355dc39f6307cc98fbd8cea314bdd381c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          9.0.4 (Oracle Corporation 9.0.4+11)
OS:           Windows 10 10.0 amd64
```

**Comment by HoldYourWaffle** [comment]
Did some further investigation, this seems to be an issue with Java 9. It works absolutely fine when testing on java 8.

**Comment by Grundlefleck** [comment]
Hey @HoldYourWaffle, thanks for reporting!

I mistakenly thought I had released a version compatible with Java 9, following #103, but now I realise that version doesn't exist. I've released v0.10.1 tonight, and it should be in Maven Central within hours. Could you try again with that and see if that fixes your issue?

Thanks again.

**Comment by HoldYourWaffle** [comment]
Everything seems to work perfectly now thanks!
###### I wonder how this has gone unnoticed for 4 months though... 🤔
